test/unit_tests/braket/default_simulator/test_density_matrix_simulator.py

```diff
@@ -0,0 +1,426 @@
+# Copyright 2019-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
This is a good point. DensityMatrixSimulator is not supposed to use StateVectorSimulation. Do you have any idea to prevent this? 

For now, the easiest way seems to be to add a check that only allows DensityMatrixSimulator to use DensityMatrixSimulation, and StateVectorSimulator to use StateVectorSimulation. When there are more simulation methods for each simulator, we can add attributes and method to allow simulators to use different simulation methods.

Reply:
Adding tests here specific to the type of simulation that the simulator performs might help. For example, a test for noise simulation. Ideally, you'd want a set of tests which capture the complete functionality you expect from the class, and not tie it to the internal implementation (if possible).

Reply:
Thanks, but my point is, in a fundamental level, density matrix simulator don't use state vector simulation. But the reverse is valid, state vector simulator can use density matrix simulation because it's more general.